### DISCUSSION TITLE
Anyone knows a fork/example with something similar to ListView/DataGrid ?

### DISCUSSION THREAD

**Discussion by iksi4prs** [spectreconsole/spectre.console#1206] [Q&A] [answered]
Hi,
Anyone knows a fork/example with something similar to ListView/DataGrid ?
With some basic functionalities, eg navigating rows, highlight current selected row, data binding, etc.
Thanks for any help.
Si.

**Comment by patriksvensson** [comment, marked as answer]
I would recommend that you check out Terminal.Gui which might be what you're looking for: https://github.com/gui-cs/Terminal.Gui

**Reply by iksi4prs** [reply]
@patriksvensson , I'm familiar with Terminal.Gui. I'm looking for the functionalities of Terminal.Gui, but with the ui/style capabilities of Spectre.

**Reply by patriksvensson** [reply]
@iksi4prs I understand, but we're not aiming at being a TUI library in that sense.

**Comment by iksi4prs** [comment]
@patriksvensson 

>  I understand, but we're not aiming at being a TUI library in that sense.

Sure, I understood from your first reply.

I've managed to do what I want using Layout(s), it was quite simple.
There's one thing that looks bit more complicated. I want to add some "popup dialogs".
I didn't find a way to do it with existing classes (I missed anything?).
It is supposed to be quite simple to get buffer data via win32 api, and restore it after closing the "popup",
but from what I see the "core" version is not implementing this, and I want my app to be cross-platform.

I thought to hook/intercept all calls to Console.Write and keep a buffer to do it,
I searched the code for a good place to do it, and while doing so I encountered "Recorder" class.
It seem do exactly this, but I couldn't find any examples or usage in code.

Is 'Recorder' what I need ?
10x for your help.

**Comment by patriksvensson** [comment]
The `Recorder` just records the `IRenderable`s that's been written to the AnsiConsole. It does not keep a reference to the output buffer. I do not know how you can solve your problem, sorry.
